Command/Helper/DoctrineCommandHelper.php

```diff
@@ -32,10 +33,12 @@ public static function setApplicationHelper(Application $application, InputInter
 
         $managerNames = $doctrine->getManagerNames();
 
-        if ($input->getOption('db') !== null || count($managerNames) === 0) {
-            self::setApplicationConnection($application, (string) $input->getOption('db'));
-        } else {
-            self::setApplicationEntityManager($application, (string) $input->getOption('em'));
+        $db = $input->getOption('db');
+        $em = $input->getOption('em');
+        if (is_string($db) || count($managerNames) === 0) {
+            self::setApplicationConnection($application, (string) $db);
```

Comment:
Do not cast `$db` to string.
When project doesn't use `doctrine/orm` it will produce error `Doctrine ORM Connection named "" does not exist.`

Reply:
The `is_string` prevents that, but it's still wrong. The error stems from a documentation mistake in DoctrineBundle, which specifies the arguments as `string` where the underlying methods accept `string|null`. I'll fix that in a 1.11 patch release and ignore the error in the meantime